crates/net/network/src/session/handle.rs

```diff
@@ -24,11 +21,20 @@ use tokio::{
 #[derive(Debug)]
 pub(crate) struct PendingSessionHandle {
     /// Can be used to tell the session to disconnect the connection/abort the handshake process.
-    pub(crate) _disconnect_tx: oneshot::Sender<()>,
+    pub(crate) disconnect_tx: mpsc::Sender<()>,
```

Comment:
why do we need to make this an mpsc?

what additional messages besides a disconnect need to be sent?

Reply:
same q

Reply:
I ran into a difficulty using the oneshot channel, let me know if you have any suggestion to workaround this in a cleaner way:
The send method of the oneshot takes ownership of it, which is inside the `PendingSessionHandle`. When calling the `on_handle_message` method of the network manager, a mutable reference of it is received, and this unables the send method to be used on the pending session handle, which is inside the sessions manager.

The mpsc channel does not have this issue, so I tried that for now altough it does not seem correct.

Reply:
I see, 
in this case, simply use an `Option` and ue `if let Some() disconnect_tx.take()`